src/drivers/legos/driver.rs

```diff
@@ -0,0 +1,402 @@
+                events.push(Event::Button(ButtonEvent::RPadTap(BinaryInput {
+                    pressed: state.rpad_tap,
+                })));
+            }
+            if state.thumb_l != old_state.thumb_l {
+                events.push(Event::Button(ButtonEvent::ThumbL(BinaryInput {
+                    pressed: state.thumb_l,
+                })));
+            }
+            if state.thumb_r != old_state.thumb_r {
+                events.push(Event::Button(ButtonEvent::ThumbR(BinaryInput {
+                    pressed: state.thumb_r,
+                })));
+            }
+
+            // Axis events
+            if state.l_stick_x != old_state.l_stick_x || state.l_stick_y != old_state.l_stick_y {
+                events.push(Event::Axis(AxisEvent::LStick(JoyAxisInput {
+                    x: state.l_stick_x,
+                    y: state.l_stick_y,
+                })));
+            }
+            if state.r_stick_x != old_state.r_stick_x || state.r_stick_y != old_state.r_stick_y {
+                events.push(Event::Axis(AxisEvent::RStick(JoyAxisInput {
+                    x: state.r_stick_x,
+                    y: state.r_stick_y,
+                })));
+            }
+
+            if state.a_trigger_l != old_state.a_trigger_l {
+                events.push(Event::Trigger(TriggerEvent::ATriggerL(TriggerInput {
+                    value: state.a_trigger_l,
+                })));
+            }
+            if state.a_trigger_r != old_state.a_trigger_r {
+                events.push(Event::Trigger(TriggerEvent::ATriggerR(TriggerInput {
+                    value: state.a_trigger_r,
+                })));
+            }
+            //TODO: When touchpad firmware is updated to use ABS events, enable this
+            //if state.touch_x != old_state.touch_x
+            //    || state.touch_y != old_state.touch_y
+            //    || state.rpad_touching != old_state.rpad_touching
+            //{
+            //    events.push(Event::Axis(AxisEvent::Touchpad(TouchAxisInput {
+            //        x: state.touch_x,
+            //        y: state.touch_y,
+            //        index: 0,
+            //        is_touching: state.rpad_touching,
+            //    })))
+            //}
+        };
+
+        events
+    }
+
+    fn handle_inertial_report(
+        &mut self,
+        buf: [u8; INERTIAL_PACKET_SIZE],
+    ) -> Result<Vec<Event>, Box<dyn Error + Send + Sync>> {
+        let input_report = InertailInputDataReport::unpack(&buf)?;
+
+        // Print input report for debugging
+        //log::debug!("--- Input report ---");
+        //log::debug!("{input_report}");
+        //log::debug!(" ---- End Report ----");
+
+        let report_type = match input_report.report_id {
+            1 => ReportType::AccelData,
+            2 => ReportType::GyroData,
+            _ => {
+                let _report_id = input_report.report_id;
+                return Err("Unknown report type: {_report_id}".into());
```

Comment:
```suggestion
                let report_id = input_report.report_id;
                return Err(format!("Unknown report type: {report_id}").into());
```

Reply:
Clippy complains about this for some reason

Reply:
It complains because `format!()` isn't being used to include the `report_id` in the error message.